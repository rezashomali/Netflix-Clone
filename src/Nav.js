import React, { useEffect, useState } from "react";
import "./Nav.css";

const Nav = () => {
  const [show, setShow] = useState(false);

  useEffect(() => {
    window.addEventListener("scroll", () => {
      if (window.scrollY > 100) {
        setShow(true);
      } else {
        setShow(false);
      }
    });
    return () => {
      window.removeEventListener("scroll");
    };
  }, []);

  return (
    <div className={`nav ${show && "nav__black"}`}>
      <img
        className="nav__logo"
        src={require("./assets/images/netflix-logo.png")}
        alt="Netflix Logo"
      />
      <img
        className="nav__avatar"
        src={require("./assets/images/user.png")}
        alt="User Logo"
      />
    </div>
  );
};

export default Nav;
