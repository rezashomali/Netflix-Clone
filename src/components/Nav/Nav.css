.nav {
  position: fixed;
  top: 0;
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 10px 20px;
  box-sizing: border-box;
  transition-timing-function: ease-in;
  transition: all 0.5s;
}

.nav__black {
  background-color: #111;
}

.nav__logo {
  width: 80px;
  object-fit: contain;
}

.nav__avatar {
  width: 30px;
  object-fit: contain;
}
