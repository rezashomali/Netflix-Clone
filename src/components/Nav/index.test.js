import React from "react";
import renderer from "react-test-renderer";
import Nav from "./index";

test("Nav snapshot ", () => {
  const component = renderer.create(<Nav />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
