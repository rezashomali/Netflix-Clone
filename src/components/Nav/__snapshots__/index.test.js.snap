// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot  1`] = `
<div
  className="nav false"
>
  <img
    alt="Netflix Logo"
    className="nav__logo"
    src="netflix-logo.png"
  />
  <img
    alt="User Logo"
    className="nav__avatar"
    src="user.png"
  />
</div>
`;
